### PR TITLE
remove limit order from liquidity

### PR DESCRIPTION
**Pull request by nlordell**
This PR removes `LimitOrder` from the `Liquidity` enum. This was done as a first step towards refactoring to make it easier to add new liquidity such that:
- `Liquidity` type will eventually be all the baseline liquidity that we fetch. This will hopefully allow us to simplify the whole solver liquidity vs API liquidity issue and make it easier to compute baseline solutions in the orderbook API for bad token detection, etc.
- We were grouping limit orders and AMM liquidity only to split it back up again in _every_ solver: `([LimitOrder], [Amms]) -> Liquidity -> ([LimitOrder], [Amms])`
- If we add actual AMM limit orders in the future (such as 0x orders) then we would need to distinguish them between the user orders.

### Test Plan

Mostly refactored for new split, should be caught by existing CI.

### CODE REVIEW

**Review by vkgnosis — APPROVED** [review]

File: solver/src/solver/baseline_solver.rs [hunk 1]
```diff
@@ -141,30 +146,29 @@ impl BaselineSolver {
...
+                                tokens: order.tokens,
+                                order: AmmOrder::ConstantProduct(order),
                             });
                         }
+                        Liquidity::WeightedProduct(order) => {
+                            for tokens in order.token_pairs() {
+                                amm_map.entry(tokens).or_default().push(Amm {
+                                    tokens,
+                                    order: AmmOrder::WeightedProduct(order.clone()),
+                                });
```
Comment (bh2smith) [inline solver/src/solver/baseline_solver.rs:166]: order is cloned in one place but not the other...

File: solver/src/solver/http_solver.rs [hunk 2]
```diff
@@ -123,18 +123,17 @@ impl HttpSolver {
...
-                Liquidity::ConstantProduct(amm) => {
-                    vec![amm.tokens.get().0, amm.tokens.get().1]
-                }
-                Liquidity::WeightedProduct(amm) => amm.reserves.keys().cloned().collect(),
-            })
+            .flat_map(|order| [order.sell_token, order.buy_token]);
+        let liquidity_tokens = liquidity.iter().flat_map(|liquidity| match liquidity {
+            Liquidity::ConstantProduct(amm) => amm.tokens.into_iter().collect::<Vec<_>>(),
+            Liquidity::WeightedProduct(amm) => amm.reserves.keys().cloned().collect(),
+        });
```
Comment (bh2smith) [inline solver/src/solver/http_solver.rs:133]: Nice.

File: solver/src/solver/http_solver.rs [hunk 3]
```diff
@@ -322,18 +322,18 @@ impl HttpSolver {
...
+            &mut orders,
+            &constant_product_liquidity,
             &self.native_token,
         );
-        let limit_orders = self.map_orders_for_solver(orders.0);
-        let constant_product_orders = self.map_amm_orders_for_solver(orders.1);
-        let weighted_product_orders = self.map_amm_orders_for_solver(orders.2);
+        let limit_orders = self.map_orders_for_solver(orders);
+        let constant_product_orders = self.map_amm_orders_for_solver(constant_product_liquidity);
+        let weighted_product_orders = self.map_amm_orders_for_solver(weighted_product_liquidity);
```
Comment (bh2smith) [inline solver/src/solver/http_solver.rs:336]: Also much nicer.

File: solver/src/solver/http_solver.rs [hunk 4]
```diff
@@ -449,28 +449,19 @@ impl HttpSolver {
...
-    (
-        limit_orders,
-        constant_product_orders,
-        weighted_product_orders,
-    )
+    (constant_product_orders, weighted_product_orders)
 }
 
+// TODO: This currently does **NOT** consider balancer pools, but definitely should.
 fn remove_orders_without_native_connection(
```
Comment (bh2smith) [inline solver/src/solver/http_solver.rs:465]: That is weird... how is this software already running fine on weighted pools if they aren't here? Is this related to price estimation? Should we link this TODO in #1075?

**Review by bh2smith — APPROVED** [review]

File: solver/src/solver/http_solver.rs (hunk 4 above)
Comment (nlordell) [reply solver/src/solver/http_solver.rs:465]: I already created a separate issue in https://github.com/gnosis/gp-v2-services/issues/1071

This was probably something that slipped by in the initial Balancer V2 weighted pool integration.

File: solver/src/solver/baseline_solver.rs (hunk 1 above)
Comment (nlordell) [reply solver/src/solver/baseline_solver.rs:166]: This was already the case before (look at line 160 that got removed).

Additionally, this is needed here. The reason is you have 1 token pair per order for constant product orders, so no `clone` required, however you have n token pairs per order for weighted products, so you need to clone `n` times. Also worth noting, clippy should report on superfluous `clone` calls, so the linter should catch unnecessary clones.
